src/utils/telemetry.ts

```diff
@@ -0,0 +1,168 @@
+import os from 'os';
+import { version } from '../../package.json';
+import path from 'path';
+import fs from 'fs/promises';
+import crypto from 'crypto';
+import { logger } from './logger';
+
+export async function isFile(path: string): Promise<boolean> {
+  try {
+    const stat = await fs.stat(path);
+    return stat.isFile();
+  } catch {
+    return false;
+  }
+}
+
+/** @type {boolean} See envvar.md. */
```

Comment:
since this is TypeScript, we can replace JSDoc type annotations with TypeScript types (but the rest of the doc is nice!).

Reply:
I sorta wanna keep the copy pasta from the CLI in this file so it's more of a port and not a rewrite. Although I do agree with most of your feedback